src/zip.c

```diff
@@ -851,22 +854,38 @@ int zip_extract(const char *zipname, const char *dir,
       goto out;
     }
 
-    if (!mz_zip_reader_is_file_a_directory(&zip_archive, i)) {
-      if (!mz_zip_reader_extract_to_file(&zip_archive, i, path, 0)) {
-        // Cannot extract zip archive to file
+    if ((((info.m_version_made_by >> 8) == 3) || ((info.m_version_made_by >> 8) == 19)) // if zip is produced on Unix or macOS (3 and 19 from section 4.4.2.2 of zip standard)
+        && info.m_external_attr & (0x20 << 24)) { // and has sym link attribute (0x80 is file, 0x40 is directory)
+#if defined(_WIN32) || defined(__WIN32__) || defined(_MSC_VER) ||              \
+    defined(__MINGW32__)
+#else
+      char symlinkTo[MAX_PATH + 1] = {0};
```

Comment:
Could you move this declaration to the top and `memset`, just to satisfy old compilers?